### PR TITLE
[REF] Upgrade DOMPDF to 2.x branch

### PR DESCRIPTION
**Pull request by seamuslee001**
Overview
----------------------------------------
This Upgrades DOMPDF to be on the 2.x branch which is the next version up and updates the cleanup script appropriately

Before
----------------------------------------
DOMPDF version 1.2.1 used

After
----------------------------------------
DOMPDF version 2.0.0 used

Technical Details
----------------------------------------
There is nothing that immediately stands out as issues for us on the release notes https://github.com/dompdf/dompdf/releases

ping @demeritcowboy @eileenmcnaughton @totten @kcristiano

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Shall we let the rc fork first to give this one that bit longer in rc?

**Comment by demeritcowboy** [comment]
Hehe you beat out dependabot!
I should probably try this out, and with the fonts changes.

**Comment by eileenmcnaughton** [comment]
Yeah dependabot's intervention suggests don't wait - merge now

**Comment by eileenmcnaughton** [comment]
OK - so I guess re security - our policy is to merge third party security releases to the RC I think - and then we port to stable as appropriate (generally if we determine the security issue does affect us).

I did the mimimum r-run - ie checked a pdf would still render

**Comment by eileenmcnaughton** [comment]
Ah - I did get an enotice in my test is seems - this is on the PR test site

image

**Comment by seamuslee001** [comment]
@eileenmcnaughton I think I have solved that e-notice now, something something to do with the font cache

**Comment by demeritcowboy** [comment]
I tested an upgrade with an existing custom font dir. Looks ok.

File: tools/scripts/composer/dompdf-cleanup.sh [hunk 1]
```diff
@@ -23,80 +23,78 @@ function safe_delete() {
 ##############################################################################
 ## usage: make_font_cache > font-cache.php
 function make_font_cache() {
-cat <<EOFONT
-<?php return array (
-  'sans-serif' =>
+php -r "echo json_encode(array (
+  'sans-serif' =>
```
Comment (demeritcowboy) [inline tools/scripts/composer/dompdf-cleanup.sh:27]: Note there is extra whitespace here at the end, and on lines 34 and 41.

Comment (demeritcowboy) [reply tools/scripts/composer/dompdf-cleanup.sh:27]: The whitespace is still there but I'll put up a PR for it.

**Pull request merged**